api/src/routes/oauth2/authorize.ts

```diff
@@ -0,0 +1,83 @@
+import { Router, Request, Response } from "express";
+import { route } from "@fosscord/api";
+import { Application, User, Member, Role, Guild } from "@fosscord/util";
+const router = Router();
+
+router.get("/", route({}), async (req: Request, res: Response) => {
+	
+    const { client_id, scope } = req.query;
+
+    const application = await Application.findOneOrFail({ where: `"id" = ${client_id}` })
+
+    const user = await User.findOneOrFail({ where: `"id" = ${req.user_id}`})
+
+    const bot = await User.findOneOrFail({ where: `"id" = ${client_id}`})
+    
+    const guilds = await Member.find({ relations: ["guild"], where: { id: req.user_id } });
+
+    let permissionGuilds = []
+
+    for (const guild of guilds.map((x) => x.guild)) {
+
+        const guild_id = guild.id
+
+        let permissions: number = 0;
+		
+        const member = await Member.findOneOrFail({
+				where: { id: req.user_id, guild_id },
+				relations: ["roles"]
+			})
+
+        for (const role of member.roles) {
+            permissions += Number(role.permissions)
+        }
+
+        if (guild.owner_id == req.user_id) {
+            permissions += 2196771451326;
```

Comment:
Maybe add a comment saying this grants admin ( plus a few other things, which I don't know why they're here, why not just grant admin? )

Reply:
this forcibly grants admin to *any app added by guild owner*, which is not desired

Reply:
no this forces admin only for guild owner which is different than the everyone role